### PR TITLE
updated docstring of stats.linregress

### PR DESCRIPTION
**Pull request by mherkazandjian**
- fixed minor style issue in docstring
- added plotting the displaying the fit line

### CODE REVIEW

**Comment by ev-br** [comment]
Good idea! 
The plot does not render though:
https://843-1460385-gh.circle-artifacts.com/0//home/ubuntu/scipy/doc/build/html-scipyorg/generated/scipy.stats.linregress.html#scipy.stats.linregress

You likely need a blank line between text and code.

**Comment by mherkazandjian** [comment]
fixed. Checked the rendered doc page locally. looks fine now.

File: scipy/stats/_stats_mstats_common.py [hunk 1]
```diff
@@ -44,17 +44,27 @@ def linregress(x, y=None):
 
     Examples
     --------
+    >>> from numpy import random
     >>> from scipy import stats
-    >>> np.random.seed(12345678)
-    >>> x = np.random.random(10)
-    >>> y = np.random.random(10)
```
Comment (ev-br) [inline scipy/stats/_stats_mstats_common.py]: nitpick: this should not be needed, `import numpy as np` is implicit.

File: scipy/stats/_stats_mstats_common.py [hunk 2]
```diff
@@ -44,17 +44,27 @@ def linregress(x, y=None):
...
     >>> print("r-squared:", r_value**2)
     ('r-squared:', 0.080402268539028335)
 
+    Plot the data along with the fitted line
+
+    >>> from matplotlib import pyplot as plt
+    >>> plt.plot(x, y, 'o', label='Original data', markersize=10)
+    >>> plt.plot(x, intercept + slope*x, 'r', label='Fitted line')
+    >>> plt.xlim(-1, 4)
+    >>> plt.ylim(-2, 5)
```
Comment (ev-br) [inline scipy/stats/_stats_mstats_common.py]: Why manually changing the plot limits? Both x and y are within [0, 1], so why having a data blob in a sea of white space?

**Comment by ev-br** [comment]
Something is still not quite right: 

https://845-1460385-gh.circle-artifacts.com/0//home/ubuntu/scipy/doc/build/html-scipyorg/generated/scipy.stats.linregress.html#scipy.stats.linregress

does not show the plot.

**Comment by ev-br** [comment]
Also at some point you'll be asked to squash the commits into one, if you know how do it.

File: scipy/stats/_stats_mstats_common.py (hunk 1 above)
Comment (mherkazandjian) [inline scipy/stats/_stats_mstats_common.py]: ok. no problem. i'll remove that.

File: scipy/stats/_stats_mstats_common.py (hunk 2 above)
Comment (mherkazandjian) [inline scipy/stats/_stats_mstats_common.py]: the plot looks better this way. but i  guess to save some lines of code i'll remove them. the default range is good enough for this example.

**Comment by mherkazandjian** [comment]
i am familiar with squashing locally. but not sure how that would work on github.... i'll check it out.

**Comment by ev-br** [comment]
> i am familiar with squashing locally. but not sure how that would work on github.... i'll check it out.

You just squash it locally (make sure to have a backup branch though) and then simply force-push: `$ git push origin linregress -f`

**Comment by mherkazandjian** [comment]
cool. tnx.

**Comment by mherkazandjian** [comment]
squashed all the commits :)

**Comment by ev-br** [comment]
Merged, thanks! And congratulations with what I believe is your first scipy commit. Keep them coming!

**Pull request merged**